### ISSUE TITLE
Package request: dashy

### ISSUE THREAD

**Issue by sburris0** [NixOS/nixpkgs#246137]
"Dashy helps you organize your self-hosted services by making them accessible from a single place"

I self-host many services with NixOS and would like to setup a nice dashboard/homepage that links to them all so that I don't have to remember any ports. I was a bit surprised that dashy didn't have a package already, as it is extremely popular on github.

**Metadata**

* homepage URL: https://dashy.to/
* source URL: https://github.com/Lissy93/dashy
* license: MIT
* platforms: should be any, it is a nodejs app

There are instructions to deploy with docker, but also instructions to [deploy from source](https://github.com/Lissy93/dashy)

**Comment by midnightveil** [comment]
I found a [version that someone seems to have made already](https://github.com/LongerHV/nixos-configuration/blob/master/overlay/dashy.nix).

**Comment by yorickvP** [comment]
I also have one https://git.yori.cc/yorick/dotfiles/src/branch/master/pkgs/dashy.nix

It's a bit annoying because it likes to run `yarn build` at runtime when you change the config, even though I'm not sure it needs to. It also depends on nodejs 16, which is EOL

**Comment by flyfloh** [comment]
Looks like it was packaged in #349149 and a module added in #355979

**Comment by sburris0** [comment]
Yep, looks like it. Closing this issue.

**Comment by TheRealGramdalf** [comment]
Hey everyone,
Turns out I missed this package request when I made the initial PRs, so nobody got a mention - sorry everyone and thanks @flyfloh for catching that!

I thought I should make note that the package is currently broken - TL;DR is no settings ever actually get applied. I'm making a PR to fix this, which I'll link to this PR.

> It's a bit annoying because it likes to run `yarn build` at runtime when you change the config, even though I'm not sure it needs to. It also depends on nodejs 16, which is EOL

@yorickvP - This is actually what the docker image upstream does - it runs a node server that serves the static webapp, which gets rebuilt whenever you make changes. In the current package I've diverged from that, instead opting to build the static webapp once at build, and serve it with nginx rather than running a full node server - there's more information in the initial PR (with some links to the relevant docs upstream) if you want to take a closer look.
The primary benefit of this approach is that it's much more lightweight, since a full node server typically takes upwards of 1.3GB (based on when I was running dashy in docker), compared to the ~10 MB it takes to run nginx. In the future I'd like to support running a node server for those that want it (some features such as widgets are unavailable without it), so I'll likely reference your module for that. Is [this](https://git.yori.cc/yorick/dotfiles/src/commit/4adbb2de9ee4d515481f998bd10344258f22c774/nixos/modules/dashy.nix) the most recent version of the module, or were there later revisions?

Lastly regarding node 16, I'm sticking with `nodePackages.nodejs`, which currently points to 20.18.0. I'm not sure if that will change when it comes to running the full server or not.
